rust_src/src/windows.rs

```diff
@@ -2017,4 +2017,49 @@ pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected)
     window.pixel_height_before_size_change
 }
 
+/// Set width and type of scroll bars of window WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+///
+/// Second parameter WIDTH specifies the pixel width for the vertical scroll
+/// bar.  If WIDTH is nil, use the scroll bar width of WINDOW's frame.
+/// Third parameter VERTICAL-TYPE specifies the type of the vertical scroll
+/// bar: left, right, nil or t where nil means to not display a vertical
+/// scroll bar on WINDOW and t means to use WINDOW frame's vertical scroll
+/// bar type.
+///
+/// Fourth parameter HEIGHT specifies the pixel height for the horizontal
+/// scroll bar.  If HEIGHT is nil, use the scroll bar height of WINDOW's
+/// frame.  Fifth parameter HORIZONTAL-TYPE specifies the type of the
+/// horizontal scroll bar: bottom, nil, or t where nil means to not display
+/// a horizontal scroll bar on WINDOW and t means to use WINDOW frame's
+/// horizontal scroll bar type.
+///
+/// Return t if scroll bars were actually changed and nil otherwise.
+#[lisp_fn(name = "set-window-scroll-bars", min = "1")]
+pub fn set_window_scroll_bars_lisp(
+    window: LispWindowLiveOrSelected,
+    width: LispObject,
+    vertical_type: LispObject,
+    height: LispObject,
+    horizontal_type: LispObject,
+) -> bool {
+    let mut window: LispWindowRef = window.into();
+    let updated_window = unsafe {
+        set_window_scroll_bars(
+            window.as_mut(),
+            width,
+            vertical_type,
+            height,
+            horizontal_type,
+        )
+    };
+
```

Comment:
Perhaps flip the following logic?

```
if updated_window.is_null() {
    false
} else {
    unsafe { apply_window_adjustment(updated_window) };
    true
}
```

Reply:
Good point, this is more readable to me than `!updated_window.is_null()`.

I updated the PR